tests/integration_tests/performance/test_net_tput.py

```diff
@@ -0,0 +1,290 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests the network throughput overhead added by Firecracker."""
+
+import json
+import time
+import statistics
+from subprocess import run
+import pytest
+
+import host_tools.network as net_tools
+
+RUNS = 1
+
+IPERF = 'iperf3'
+ITERATION_DURATION = 1
+WINDOW_SIZE = 10
+
+# BW in Mbits.
+BANDWIDTHS = {
+    'udp': [100, 5000, 10000],
+    'tcp': [5000, 10000, 20000, 0]
```

Comment:
The only case I see where it makes sense for tcp bandwith parameter is if we take the cpu usage of firecracker from host or if we check the retransmissions. For example we want to have 1Gbps with 10% CPU usage and 0 retransmissions.
If you don't plan to get this tcp bandwith parameter doesn't make sense, since we want to get the peak throughput

Reply:
I am going to introduce CPU usage as a next step since even iperf3 offers some details about CPU usage, so it would be an easy change.

Reply:
I don't think CPU usage from iperf would help. Here you need to take CPU usage of the emulation thread from Firecracker. You need to take this from the host.

Reply:
Actually I think it helps. The thing is that Firecracker uses 3 threads: firecracker, fc_api and fc_vcpu 0(supposing we spawn a microvm with one vcpu). The emulation usage is summed up with the vcpu usage in the firecracker thread, so I think the emulation cost would be computed in this way :  

`emulation_cpu_usage = firecracker_cpu_usage - iperf_cpu_usage`

That's why I was saying the iperf3 info about the cpu usage would be useful(of course if I'm not missing smth).

Reply:
In fact is this way - firecracker_cpu_usage - iperf_cpu_usage = guest_kernel_cpu + other small stuff + emulation. If there is no technical way for getting the emulation thread usage is fine to assume that the small stuff is small and the guest kernel cpu usage is constant related to the iperf test. But here you can take using htop the cpu usage for emulation thread.